widgets/common/src/ExecuteDeploymentModal.tsx

```diff
@@ -111,6 +112,11 @@ export default function ExecuteDeploymentModal({
         clearErrors();
     }, [dryRun, force, schedule, scheduledTime]);
 
+    let deploymentsList = deployments;
+    if (_.isEmpty(deployments)) {
+        deploymentsList = _.compact([deploymentId]);
+    }
```

Comment:
Minor:

```suggestion
    const deploymentsList: string[] = _.isEmpty(deployments) ? _.compact([deploymentId]) : deployments;
```

has the benefit of not making `deploymentsList` a mutable variable :slightly_smiling_face: 

Also, adding a `string[]` type adds a bit of documentation to the code